projects/ad7606x_fmc/common/ad7606x_bd.tcl

```diff
@@ -1,125 +1,218 @@
-  ad_ip_instance clk_wiz adc_clk_generator
-  ad_ip_parameter adc_clk_generator CONFIG.PRIMITIVE PLL
-  ad_ip_parameter adc_clk_generator CONFIG.RESET_TYPE ACTIVE_LOW
-  ad_ip_parameter adc_clk_generator CONFIG.USE_LOCKED false
-  ad_ip_parameter adc_clk_generator CONFIG.CLKOUT1_REQUESTED_OUT_FREQ 100.000
-  ad_ip_parameter adc_clk_generator CONFIG.CLKOUT1_REQUESTED_PHASE 0.000
-  ad_ip_parameter adc_clk_generator CONFIG.CLKOUT1_REQUESTED_DUTY_CYCLE 50.000
-  ad_ip_parameter adc_clk_generator CONFIG.PRIM_SOURCE Global_buffer
-  ad_ip_parameter adc_clk_generator CONFIG.CLKIN1_UI_JITTER 0
-  ad_ip_parameter adc_clk_generator CONFIG.PRIM_IN_FREQ 100.000 
-
-  ad_connect  sys_cpu_clk adc_clk_generator/clk_in1
-  ad_connect  sys_cpu_resetn adc_clk_generator/resetn
-  ad_connect  adc_clk_generator/clk_out1 axi_ad7606x/external_clk
-} else {
-  ad_connect  sys_cpu_clk axi_ad7606x/external_clk
-}
-
-# interface connections
-
-ad_connect  rx_db_o axi_ad7606x/rx_db_o
-ad_connect  rx_db_i axi_ad7606x/rx_db_i
-ad_connect  rx_db_t axi_ad7606x/rx_db_t
-ad_connect  rx_rd_n axi_ad7606x/rx_rd_n
-ad_connect  rx_wr_n axi_ad7606x/rx_wr_n
-
-ad_connect  rx_cs_n axi_ad7606x/rx_cs_n
-ad_connect  rx_cnvst_n axi_pwm_gen/pwm_0
-ad_connect  rx_busy axi_ad7606x/rx_busy
-ad_connect  rx_first_data axi_ad7606x/first_data
-
-ad_connect  sys_cpu_clk axi_ad7606x_dma/s_axi_aclk
-ad_connect  sys_cpu_clk axi_pwm_gen/s_axi_aclk
-ad_connect  sys_cpu_resetn axi_pwm_gen/s_axi_aresetn
-
-ad_connect  axi_ad7606x/adc_clk ad7606x_adc_pack/clk
-ad_connect  axi_ad7606x/adc_clk axi_ad7606x_dma/fifo_wr_clk
-ad_connect  axi_ad7606x/adc_reset ad7606x_adc_pack/reset
-ad_connect  axi_ad7606x/adc_valid ad7606x_adc_pack/fifo_wr_en
-ad_connect  ad7606x_adc_pack/packed_fifo_wr axi_ad7606x_dma/fifo_wr
-ad_connect  ad7606x_adc_pack/fifo_wr_overflow axi_ad7606x/adc_dovf
-
-for {set i 0} {$i < 8} {incr i} {
-  ad_connect axi_ad7606x/adc_data_$i ad7606x_adc_pack/fifo_wr_data_$i
-  ad_connect axi_ad7606x/adc_enable_$i ad7606x_adc_pack/enable_$i
+switch $INTF {
+  0 {
+    # data, read and write lines
+    create_bd_port -dir O -from 15 -to 0 rx_db_o
+    create_bd_port -dir I -from 15 -to 0 rx_db_i
+    create_bd_port -dir O rx_db_t
+    create_bd_port -dir O rx_rd_n
+    create_bd_port -dir O rx_wr_n
+
+    # control lines
+    create_bd_port -dir O rx_cs_n
+    create_bd_port -dir I rx_first_data
+
+    # instantiation
+    ad_ip_instance axi_ad7606x axi_ad7606x
+    ad_ip_parameter axi_ad7606x CONFIG.DEV_CONFIG $DEV_CONFIG
+    ad_ip_parameter axi_ad7606x CONFIG.ADC_N_BITS $ADC_N_BITS
+    ad_ip_parameter axi_ad7606x CONFIG.ADC_TO_DMA_N_BITS $ADC_TO_DMA_N_BITS
+    ad_ip_parameter axi_ad7606x CONFIG.EXTERNAL_CLK $EXT_CLK
+
+    # axi_pwm_gen
+    ad_ip_parameter ad7606_pwm_gen CONFIG.ASYNC_CLK_EN 0
+    ad_ip_parameter ad7606_pwm_gen CONFIG.N_PWMS 1
+    if {$DEV_CONFIG == 0} { 
```

Comment:
trailing whitespace, there is in other places, so I will provide a general solution.
git diff/show should highlight trailing whites space, and you can do:
```
git diff a0f8990ca6e64c8b1e8f6665940d0479a81f3c0c~..@ --name-only --diff-filter=dr | xargs sed -i -E 's/[[:space:]]*$//'
```
then commit the changes
explained:
diff from one commit before your first change until head, then pass the list of edited files to sed to remove trailing space. `--diff-filter=dr` excludes deleted files.